### ISSUE TITLE
When upgrading PHP version, mainenance:enable runs against old codebase, using new php version

### ISSUE THREAD

**Issue by edward-simpson** [davidalger/capistrano-magento2#167]
I'm working on an upgrade branch, which upgrades the magento version, and the php version.

Because of this, the deployed version of the site is 2.4.3, and php7.4, but the version I'm deploying is 2.4.6, and building using php8.1.

This means that when cap tries to put the site into maintenance, it's using php8.1 against M2.4.3, which is incompatible, so the deploy is failing

Is there a way of specifying a specific php path for a specific command?

**Comment by erikhansen** [comment]
@edward-simpson In your `tools/cap/config/deploy.rb` file, you can add a line like this:

```
# Only uncomment this if the `current` directory is broken for some reason, which would prevent Capistrano from enabling maintenance mode in that directory
set :magento_deploy_maintenance, false
```

And then comment it out whenever you don't need it.

**Comment by edward-simpson** [comment]
@erikhansen yeah that's had to be my solution for now, and manually putting the site in to maintenance mode beforehand
